source/databases-collections.txt

```diff
@@ -0,0 +1,275 @@
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: table, row, organize, storage
+
+Overview
+--------
+
+In this guide, you can learn how to interact with MongoDB databases and
+collections with the {+driver-short+}.
+
+MongoDB organizes data into a hierarchy of the following levels:
+
+- **Databases**: The top level of data organization in a MongoDB instance.
+- **Collections**: MongoDB stores documents in collections. They are analogous to tables in relational databases.
+- **Documents**: Contain literal data such as string, numbers, dates, and other embedded documents.
+
+For more information about document field types and structure, see the
+:manual:`Documents </core/document/>` guide in the {+mdb-server+} manual.
+
+Access a Database
+-----------------
+
+Access a database by using the ``mongoc_client_get_database()`` function and passing in
+a ``mongo_client_t`` instance and the name of the database as parameters.
+
+The following example accesses a database named ``"test_database"``:
+
+.. code-block:: c
+
+   mongoc_database_t *database = mongoc_client.get_database (
+       client, "test_database");
+
+Access a Collection
+-------------------
+
+Access a collection by using the ``mongoc_client_get_collection()`` or ``mongoc_database_get_collection()``
+functions.
+
+The following example accesses a collection named ``"test_collection"`` by using the 
+``mongoc_database_get_collection()`` function:
+
+.. code-block:: c
+
+   mongoc_collection_t *collection = mongoc_database_get_collection (
+       database, "test_collection");
+
+.. tip::
+
+   If the provided collection name does not already exist in the database,
+   MongoDB implicitly creates the collection when you first insert data
+   into it.
+
+Create a Collection
+-------------------
+
+Use the ``mongoc_database_create_collection()`` function to explicitly create a collection in a
+MongoDB database.
+
+The following example creates a collection called ``"example_collection"``:
+
+.. code-block:: c
+
+   bson_error_t error;
+   mongoc_collection_t *collection = mongoc_database_create_collection (
+       database, "example_collection", NULL, &error);
+
+You can specify collection options, such as maximum size and document
+validation rules, by passing them inside a ``bson_t`` structure. For a full list of
+optional parameters, see the :manual:`create documentation </reference/command/create>`
+in the {+mdb-server+} manual.
+
+Get a List of Collections
+-------------------------
+
+You can query for a list of collections in a database by calling the
+``mongoc_database_find_collections_with_opts()`` function. The function returns a cursor containing all
+collections in the database and their associated metadata.
+
+The following example calls the ``mongoc_database_find_collections_with_opts()`` function and iterates over
+the cursor to print the results:
+
+.. code-block:: c
```

Comment:
S: since you're printing cursor results in the code, I think it makes sense to include code output for this example. Maybe you could pass the "test_database" database and print the two collections used on this page (test_collection, example_collection)?